### PR TITLE
fix: docs for audit 

### PR DESCRIPTION
**Pull request by xenide**

### CODE REVIEW

File: src/GenericFactory.sol [hunk 1]
```diff
@@ -130,6 +130,7 @@ contract GenericFactory is IGenericFactory, Owned {
 
     event Pair(address indexed token0, address indexed token1, uint256 curveId, address pair);
 
+    // maps tokenA, tokenB addresses to pair address, where the order of tokenA and tokenB does not matter
```
Comment (OliverNChalk) [inline src/GenericFactory.sol]: Should be natspec?

File: src/curve/stable/StablePair.sol [hunk 2]
```diff
@@ -302,6 +294,9 @@ contract StablePair is ReservoirPair {
         int112 currLogLiq = ConstantProductOracleMath.calcLogLiq(aReserve0, aReserve1);
         prevClampedPrice = currClampedPrice;
 
+        // overflow is desired here as the consumer of the oracle will be reading the difference in those
+        // accumulated log values
+        // when the index overflows it will overwrite the oldest observation and then forms a loop
```
Comment (OliverNChalk) [inline src/curve/stable/StablePair.sol:297]: Should mention the case where we have no swaps for N years and the timestamp overflows twice

File: src/interfaces/IReservoirCallee.sol [hunk 3]
```diff
@@ -2,7 +2,9 @@
 pragma solidity ^0.8.0;
 
 interface IReservoirCallee {
+    /// @param sender the target address for the arbitrary data call in the case of a flash swap
```
Comment (OliverNChalk) [inline src/interfaces/IReservoirCallee.sol]:
```suggestion
    /// @param user that initiated the swap and is triggering the callback
```

File: src/interfaces/IReservoirCallee.sol [hunk 4]
```diff
@@ -2,7 +2,9 @@
...
     /// @param amount0 positive indicates the amount out (received by callee), negative indicates the amount in (owed by callee)
     /// @param amount1 positive indicates the amount out (received by callee), negative indicates the amount in (owed by callee)
+    /// @param data the bytes that are provided in the case of a flash swap
```
Comment (OliverNChalk) [inline src/interfaces/IReservoirCallee.sol]:
```suggestion
    /// @param data provided by the user is returned as part of the callback
```

File: src/asset-management/AaveManager.sol [hunk 5]
```diff
@@ -101,16 +101,12 @@ contract AaveManager is IAssetManager, Owned(msg.sender), ReentrancyGuard {
...
-
-    function setLowerThreshold(uint128 aLowerThreshold) external onlyOwner {
-        require(aLowerThreshold <= 1e18 && aLowerThreshold <= upperThreshold, "AM: INVALID_THRESHOLD");
+    function setThresholds(uint128 aLowerThreshold, uint128 aUpperThreshold) external onlyOwner {
+        require(aUpperThreshold <= 1e18 && aUpperThreshold >= aLowerThreshold, "AM: INVALID_THRESHOLDS");
         lowerThreshold = aLowerThreshold;
+        upperThreshold = aUpperThreshold;
         emit LowerThreshold(aLowerThreshold);
+        emit UpperThreshold(aUpperThreshold);
     }
```
Comment (OliverNChalk) [inline src/asset-management/AaveManager.sol]: Should we not collpase the logs into a single one?

Comment (xenide) [reply src/asset-management/AaveManager.sol]: good thinking, yes we should. Fixed

File: src/interfaces/IReservoirCallee.sol (hunk 3 above)
Comment (xenide) [reply src/interfaces/IReservoirCallee.sol]: fixed

File: src/interfaces/IReservoirCallee.sol (hunk 4 above)
Comment (xenide) [reply src/interfaces/IReservoirCallee.sol]: fixed

File: src/GenericFactory.sol (hunk 1 above)
Comment (xenide) [reply src/GenericFactory.sol]: fixed

File: src/curve/stable/StablePair.sol (hunk 2 above)
Comment (xenide) [reply src/curve/stable/StablePair.sol:297]: okay added the comment, but added it in ReservoirPair L147 as I think it's a more appropriate venue for this comment

**Review by OliverNChalk — APPROVED** [review]

**Pull request merged**